vaccine_feed_ingest/runners/_shared/fetch.py

```diff
@@ -35,22 +37,50 @@
     )
     raise e
 
-try:
-    logger.info(
-        "Scraping %s/arcgis into output_dir=%s, with config from %s",
-        state.upper(),
-        output_dir,
-        yml_config,
-    )
-    for service_item in config["arcgis"]:
-        if len(service_item["layer_names"]) > 0:
-            arcgis_ingest.fetch_geojson(
-                service_item["id"], output_dir, service_item["layer_names"]
-            )
-except KeyError as e:
-    logger.error(
-        "config file must have key 'arcgis' containing a list of objects, "
-        "each with a key 'id' and a key 'layer_names'. This config does not - %s",
-        yml_config,
-    )
-    raise e
+logger.info(
+    "Scraping %s/%s into output_dir=%s, with config from %s",
+    state.upper(),
+    config.get("parser", "arcgis"),
+    output_dir,
+    yml_config,
+)
+
+if "parser" not in config or config["parser"] == "arcgis":
+    try:
+        for service_item in config["arcgis"]:
+            if len(service_item["layer_names"]) > 0:
+                arcgis_ingest.fetch_geojson(
+                    service_item["id"], output_dir, service_item["layer_names"]
+                )
+    except KeyError as e:
+        logger.error(
+            "config file must have key 'arcgis' containing a list of objects, "
+            "each with a key 'id' and a key 'layer_names'. This config does not - %s",
+            yml_config,
+        )
+        raise e
+elif config["parser"] == "prepmod":
+    try:
+        base_url = "%s/appointment/en/clinic/search" % config["url"]
+        page = 1
+        while True:
+            url = "{}?page={}".format(base_url, page)
+            response = requests.get(url, allow_redirects=False)
+
+            # when out of results, will return 302
+            if response.status_code != 200:
+                break
+
+            with open(os.path.join(output_dir, "{}.html".format(page)), "w") as f:
```

Comment:
minor: f-strings are sightly easier to read for inline formatting e.g. f"{page}.html"

Reply:
Done